kit/transport/http/middleware.go

```diff
@@ -18,13 +18,18 @@ type Middleware func(http.Handler) http.Handler
 func SetCORS(next http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
 		if origin := r.Header.Get("Origin"); origin != "" {
+			// Access-Control-Allow-Origin must be present in every response
 			w.Header().Set("Access-Control-Allow-Origin", origin)
+		}
+		if r.Method == http.MethodOptions {
+			// allow and stop processing in pre-flight requests
 			w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
 			w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, Authorization, User-Agent")
+			w.WriteHeader(http.StatusNoContent)
+		} else {
```

Comment:
Instead of the `else`, could the code return early to avoid branching?

Reply:
yes, it also better to signalizes that the processing must be stopped